src/anyp/TrafficMode.h

```diff
@@ -12,13 +12,8 @@
 namespace AnyP
 {
 
-/**
- * Set of 'mode' flags defining types of traffic which can be received.
- *
- * Use to determine the processing steps which need to be applied
- * to this traffic under any special circumstances which may apply.
- */
-class TrafficMode
+/// POD representation of TrafficMode flags
+class TrafficModeFlags
```

Comment:
> Amos: It can more simply be a class with some flags and some helper method for checking complicated flag permutations.

AFAICT, in the current PR, only one port flag (accelSurrogate) can be exposed in its raw form. All others are never accessed in isolation by non-configuration code and, hence, should not be exposed. Clearly, the added complexity to replace that single flag with the accelSurrogate() getter method is tiny and worth the gained protection from accidental modification.

Another very important reason for wrapping all flags into methods is to make sure that we check/validate all flag reading code when adding a _new_ flag/state. Without these changes, authors, reviewers, and code porters would miss code that required modifications to support the new flag/state.

I suggest postponing this discussion until most other change requests are resolved. If the resulting PR code looks very different, with many methods returning a single flag, and the danger of missing important use cases in ported code somehow diminishes, then the above "there is a simpler solution" argument might become valid. Needless to say, it would be trivial to replace flag-returning getters with raw flag accesses in that hypothetical case, undoing many of the current changes. We are gaining a lot by _delaying_ that reversal until it is clear that it is actually a good idea.

P.S. I believe Eduard has [explained](https://github.com/squid-cache/squid/pull/835#discussion_r644761871) why having two classes is better, so I am not going to comment on that part of the change request.

Reply:
@eduard. I am thinking the needs can be met by just making the flags private is enough. No need for a completely separate class layer.

Reply:
>  just making the flags private is enough

I tried this approach during development but then rejected it after realizing that adding ~ 10 additional setter/getter methods (just for configuration needs!) made this simple code too clumsy.

Reply:
I still think it would be better as const members with accessors, even if verbose/kludgy. But not going to insist on this now.